### ISSUE TITLE
Training loss

### ISSUE THREAD

**Issue by sonal-bansal** [mseitzer/srgan#4]
Hi ,
I'm training srresnet on DIV2K from scratch(no pretrained) at lr: e-4 , and it is decayed by 0.1 for step=200,
but I'm having MSE loss(at 402 epoch) as:-

Epoch=402, lr=1.0000000000000002e-06 
Epoch [402/500], Step [1/50], Loss: 74.55887,Total_loss_old:74.55887  
Epoch [402/500], Step [2/50], Loss : 77.55219,Total_loss_old:152.11106  
Epoch [402/500], Step [3/50], Loss: 58.70598,Total_loss_old:210.81704  
Epoch [402/500], Step [4/50],Loss: 78.69186,Total_loss_old:289.50890  
Epoch [402/500], Step [5/50], Loss: 87.79494,Total_loss_old:377.30384  
Epoch [402/500], Step [6/50], Loss: 24.87956,Total_loss_old:402.18340  
.....
Overall Total_loss: 72.45043201446533

I dont know if actually training or not.

**Comment by mseitzer** [comment]
Hi,

it is hard to say what's going on. The high MSE loss indicates that something goes wrong. Are you using this implementation?

**Comment by sonal-bansal** [comment]
Yes.
Since I'm training this without any pretrained wgts, is this the reason?
and how much training does SRResnet take from scratch?

**Comment by sonal-bansal** [comment]
[code.zip](https://github.com/mseitzer/srgan/files/2828836/code.zip)

Please check this code once.I'm confused how to solve this.
Infact I'm getting ouput image nearly equal to target.(see this )
[result_images.docx](https://github.com/mseitzer/srgan/files/2828716/result_images.docx)

**Comment by mseitzer** [comment]
Seems to me that you get high MSE and negative PSNR because your inputs and/or the labels are not in the same scale. Check if you scale your both inputs and labels to range [0, 1].

**Comment by sonal-bansal** [comment]
Earlier I was thinking of using transform.Normalise((0,0,0,0),(1,1,1,1)), would it be ok to use it for both input and label?
Moreover ,even after applying this ...my predicted image was still out of range[0,1],and even when i converted it using this:
def scale_to_range(tensor, target_range, clamp=True):
  """Scales tensor from (0, 1) range to `target range`"""
  tensor = tensor * (target_range[1] - target_range[0]) + target_range[0]
  if clamp:
    tensor = tensor.clamp(target_range[0], target_range[1])
  return tensor

I wasnt able to plot/save my output  as it appeared like a simple black background

If possible can you send me it's code or formula to do it,which you used or tell me what all normalisation/scale required for input,label,predicted.

Thank you

**Comment by sonal-bansal** [comment]
My input and label are in scale(0,255) , and my output is coming in random scale.
I tried using ToTensor() to scale input and label to (0,1) but even after that my output is still coming in random scale and not (0,1).
Because of this I'm not able to view my desired output , and it appears like this:-
![image](https://user-images.githubusercontent.com/33753404/52328220-cf787000-2a14-11e9-9cb6-cbb9d8d5d59d.png)

Whereas I tried second approach where by input and label were(0,255) and output in random scale,
I calculated loss using:-
       #loss = criterion(outputs.type(torch.cuda.ByteTensor).float().div(255), labels.float().div(255))
With this I'm getting psnr as 31.
and output image as:
[result_images.docx](https://github.com/mseitzer/srgan/files/2828716/result_images.docx)
Please tell me what should i do.

**Comment by mseitzer** [comment]
If you use scale (0, 1) you might have to convert back to range (0, 255) to visualize the images, that is why you get black images. 

You can probably use either range (0, 1) or (0, 255) to train the images, as long as you are consistent over inputs, labels, and metrics. Range (0, 1) is preferable as it fits better to the network initialization.

Your images in the docx look fine.

**Comment by sonal-bansal** [comment]
Hi,
So, since I've trained my model for (0,255) input and label , and my output
is also in approximately around this range only, so i should calculate mse
loss in the same range (which is coming as 56 wid the output i showed you)
or i should convert them to (0,1) before displaying loss (which when tries
had a loss of 0.00461) ,

See i know i need to do loss.backward() wid the same scale only but for
displaying which scale should i use(bcz 50 seems high)??

Pnsr(0,255) =10log((255^2)/mse)
Psnr(0,1) =10 log(1/mse)

Both tried what should i use?

**Comment by mseitzer** [comment]
The answer is, the scale of the loss does not matter. Both work. Conventionally, people would use a scale of (0, 1). For comparing to others, use PSNR, which is invariant to the scale if correctly computed.
